services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/models/shared_store.py

```diff
@@ -74,6 +85,7 @@ async def _setup_initial_volume_states(self) -> None:
                 (VolumeCategory.OUTPUTS, VolumeStatus.CONTENT_NEEDS_TO_BE_SAVED),
                 (VolumeCategory.STATES, VolumeStatus.CONTENT_NEEDS_TO_BE_SAVED),
             ]:
+                # pylint: disable=unsupported-assignment-operation
```

Comment:
why is this necessary? this looks fishy

Reply:
no clue, had to use it in a couple of places @giancarloromeo any idea why?

Reply:
Static analysis tools struggle with dynamic frameworks like Pydantic. It can be safely disabled, or if you don't want to disable it, ensure that the `self.volume_states` is a `dict` (i.e. with `__setitem__` method) adding a check. But this will result more pedantic than pydantic.

```python
    volume_states: dict[VolumeCategory, VolumeState] = Field(
        default_factory=dict, description="persist the state of each volume"
    )

    async def _setup_initial_volume_states(self) -> None:
        async with self:
            for category, status in [
                (VolumeCategory.INPUTS, VolumeStatus.CONTENT_NO_SAVE_REQUIRED),
                (VolumeCategory.SHARED_STORE, VolumeStatus.CONTENT_NO_SAVE_REQUIRED),
                (VolumeCategory.OUTPUTS, VolumeStatus.CONTENT_NEEDS_TO_BE_SAVED),
                (VolumeCategory.STATES, VolumeStatus.CONTENT_NEEDS_TO_BE_SAVED),
            ]:
                if self.volume_states is None:
                    self.volume_states = {}
                self.volume_states[category] = VolumeState(status=status)

```